Modules/Bidding.lua

```diff
@@ -771,14 +771,19 @@ function CommDKP:BroadcastStopBidTimer()
 end
 
 function CommDKP_Register_ShiftClickLootWindowHook()      -- hook function into LootFrame window. All slots on ElvUI. But only first 4 in default UI.
-  local num = GetNumLootItems();
-
-  if getglobal("ElvLootSlot1") then       -- fixes hook for ElvUI loot frame
+  C_Timer.After(0.0001, function()
+    local num = GetNumLootItems();
+    local lootSlotNameTemplate = "LootButton"
+    if getglobal"ElvLootSlot1" then       -- fixes hook for ElvUI loot frame

```

Comment:
Looks like typos here.

Reply:
There was a large if statement that basically worked like this:
```lua
if ElvUIExists then -- actually checks if "ElvLootSlot1" exists, which indicates that loot frames are being overridden
  for i = 1, maxLootSlots do
    -- big block of code for hooking onto "ElvLootSlot"..i
  end
else
  for i = 1, math.min(maxLootSlots, 4) do
    -- block of identical code for hooking onto "LootButton"..i
  end
end
```
All the work for hooking was the same except for the maximum number of items being hooked (4 without ElvUI) and the name of the loot frame that's being hooked. So I simplified to this:
```lua
local lootSlotNameTemplate = "lootButton" -- assume ElvUI doesn't exist
local num = numberOfSlots() -- get the number of items on the corpse/chest/whatever
if "ElvLootSlot1" then -- just checks if there's an ElvUI loot frame in existence
  lotSlotNameTemplate = "ElvLootSlot" -- oops, we're using ElvUI. loot slot frames have a different name with ElvUI, better change it
else
  num = math.min(maxLootSlots, 4) -- ElvUI doesn't exist, so we clamp to 4
end
for i = 1, maxLootSlots do -- for loop works for ElvUI or default UI
  -- hook onto LootSlotTemplate..i -- LootSlotNameTemplate isn't even used more than once. The rest of the code for hooking is identical with or without ElvUI
end
```

Except for the C_Timer being used to delay this function by a single frame, the other changes don't have any impact. It's just refactoring, code cleanup. The function easier to understand and more maintainable. Shorter, no duplicate code.
I've tested with ElvUI on and off with these changes, and the loot frames hook correctly the very first time they're used.

Reply:
```if getglobal"ElvLootSlot1" then ```  Is missing parenthesis.